File: maple/maple-user-web/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../user/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../user/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../user/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../user/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../user/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../user/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../user/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	//resume
	this.resumemodel=function(){
		return $http.get('../user/ResumeModel.do')
	}
    this.showName=function(){
        return $http.get('../user/name.do')
    }
    //job
    this.jobmodel=function(){
        return $http.get('../user/JobModel.do')
    }
    //compy
    this.compymodel=function(){
        return $http.get('../user/CompyModel.do')
    }

    //搜索
    this.searchJob=function(page,rows,searchEntity){
        return $http.post('../user/searchJob.do?page='+page+"&rows="+rows, searchEntity);
    }
    //搜索
    this.findbusModel=function(page,rows,searchEntity){
        return $http.post('../user/searchCompy.do?page='+page+"&rows="+rows, searchEntity);
    }
    //搜索
    this.finddataModel=function(page,rows,searchEntity){
        return $http.post('../user/searchData.do?page='+page+"&rows="+rows, searchEntity);
    }

    this.finddata=function (id) {
        return $http.post('../user/datadetail.do?id='+id);
    }
    this.addtask=function(id,num){
		return $http.get('../data/addtask.do?id='+id+"&num="+num);
	}
});

File: maple/maple-type-interface/src/main/java/com/maple/service/UserService.java
package com.maple.service;
import java.util.List;

import com.maple.pojo.Compy;
import com.maple.pojo.Job;
import com.maple.pojo.User;

import com.maple.pojo.Workdata;
import entity.*;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface UserService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<User> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(User user);
	
	
	/**
	 * 修改
	 */
	public void update(User user);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public User findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(User user,int pageNum,int pageSize,String name);


	/**
	 * 更新shi否为boss
	 * @param id
	 * @param status
	 */
	public void updateRolereq(Long id,String  rolereq) ;

	public void updateStatus(Long id,String status);

	public User findoneByuserid(String userid);

	public ResumeModel findResumeModel(String name);

	public List<JobModel> findJobModel();

	public List<BusModel> findCompyModel();

	public BusModel findOneCompyModel(int id);

	public  List<JobModel> findJobModel(int id);



	public PageResult findJobPage(Job user,int pageNum,int pageSize,String name);

	public PageResult findCompyPage(Compy Compy, int pageNum, int pageSize, String name);

	public PageResult findWorkdataPage(Workdata workdata, int pageNum, int pageSize, String name);

	public DataModel findOneDataModel(int id);
	ResumeModel finddataResume(String name);
}
